File: 구름 알고리즘 미션/그리디/보조배터리/보조배터리.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

let [A, B, C] = input[0].split(' ').map(Number);
const N = +input[1];
const chargers = input.slice(2).map(line => {
  const [price, type] = line.split(' ').map(Number);
  return { price, type };
});

chargers.sort((a, b) => a.price - b.price);

let count = 0;
let total = 0;

for (const charger of chargers) {
  const { price, type } = charger;

  if (type === 0) { // X 타입
    if (A > 0) {
      A--;
    } else if (C > 0) {
      C--;
    } else {
      continue;
    }
  } else { // Y 타입
    if (B > 0) {
      B--;
    } else if (C > 0) {
      C--;
    } else {
      continue;
    }
  }

  count++;
  total += price;
}

console.log(count, total);

File: 구름 알고리즘 미션/그리디/보조배터리/보조배터리.py
import sys
input = sys.stdin.readline

A, B, C = map(int, input().split())
N = int(input())
charger = [tuple(map(int, input().split())) for _ in range(N)]

charger.sort()

cnt, cash = 0, 0

for c, t in charger:
    if t == 0:  # X 타입
        if A > 0:
            A -= 1
        elif C > 0:
            C -= 1
        else:
            continue
    else:  # Y 타입
        if B > 0:
            B -= 1
        elif C > 0:
            C -= 1
        else:
            continue

    cnt += 1
    cash += c

print(cnt, cash)
